src/Tribe/PUE/Notices.php

```diff
@@ -85,32 +113,55 @@ public function register_name( $plugin_name ) {
 	}
 
 	/**
-	 * Restores plugins added on previous requests to the relevant notification
-	 * groups.
+	 * Restores and sanitizes plugin notices to ensure data integrity and prevent memory issues.
+	 *
+	 * Retrieves saved notices from the database, validates and sanitizes them,
+	 * and sets them as the current request's notices.
+	 *
+	 * @since TBD Switched from `array_merge_recursive` to `wp_parse_args` to fix data duplication issues. Added additional sanitation and memory safeguards to handle large data sets effectively.
+	 *
+	 * @return void
 	 */
 	protected function populate() {
-		$this->saved_notices = (array) get_option( self::STORE_KEY, [] );
-
-		if ( empty( $this->saved_notices ) ) {
-			return;
-		}
+		$this->saved_notices = $this->sanitize_notices( (array) get_option( self::STORE_KEY, [] ) );
 
-		$this->notices = array_merge_recursive( $this->notices, $this->saved_notices );
+		// Init the notices as the saved notices.
+		$this->notices = $this->saved_notices;
+	}
 
-		// Cleanup
-		foreach ( $this->notices as $key => &$plugin_lists ) {
-			// Purge any elements that are not arrays
-			if ( ! is_array( $plugin_lists ) ) {
-				unset( $this->notices[ $key ] );
+	/**
+	 * Recursively sanitizes notices to prevent nesting and ensure data integrity.
+	 *
+	 * @param array $notices The array of notices to sanitize.
+	 *
+	 * @since TBD
+	 *
+	 * @return array Sanitized notices.
+	 */
+	protected function sanitize_notices( array $notices ): array {
+		foreach ( $notices as $key => &$plugin_list ) {
+			// Ensure the value is an array; otherwise, reset it.
+			if ( ! is_array( $plugin_list ) ) {
+				$plugin_list = [];
```

Comment:
One question: if it's not an array, what is it? Are we "losing data" here?

Reply:
See the test cases of how we handle that as well, but a quick answer is if its not an array for us it is now invalid and we filter it out :D